lore/io/multi_connection_proxy.py

```diff
@@ -0,0 +1,49 @@
+import random
+import logging
+import re
+
+from lore.io.connection import Connection
+from lore.util import scrub_url
+
+logger = logging.getLogger(__name__)
+
+
+class MultiConnectionProxy(object):
+
+    def __init__(self, urls, name='connection', watermark=True, **kwargs):
+        sticky = kwargs.pop('sticky_connection', None)
+        sticky = False if sticky is None else (sticky.lower() == 'true')
+
+        self._urls = urls
+        self._sticky = sticky
+        self._connections = []
+        self._active_connection = None
+
+        self.parse_connections(name, watermark, **kwargs)
+
+    def parse_connections(self, name, watermark, **kwargs):
+        kwargs.pop('url', None)
+        for url in re.split(r'\s+', self._urls):
+            c = Connection(url, name=name, watermark=watermark, **kwargs)
+            self._connections.append(c)
+        self.shuffle_connections()
+
+    def shuffle_connections(self):
+        if len(self._connections) == 0:
+            return
+        if len(self._connections) == 1:
+            self._active_connection = self._connections[0]
+        else:
+            filtered = list(filter(lambda x: x is not self._active_connection, self._connections))
+            self._active_connection = filtered[0] if len(filtered) == 1 else random.choice(filtered)
+        self.log_connection()
+
+    def log_connection(self):
+        logger.debug("using database connection {}".format(scrub_url(self._active_connection.url)))
+
+    # proxying - forward getattr to self._active_connection if not defined in MultiConnectionProxy
+
+    def __getattr__(self, attr):
```

Comment:
Wouldn't this shuffle connections even if we are just looking at the attributes of the connection? Seems like that could cause issues right?

Reply:
At first, I thought it was acceptable since shuffle_connections() is really cheap. 
But now I think you are right. We should be explicit about when we do auto-rotation (if self.__sticky is False)